lib/xray_creator.py

```diff
@@ -42,12 +42,8 @@ def _initialize_books(self):
 
         self._books = []
         for book_id in self._book_ids:
-            book = Book(self._db, book_id, self._gConnection, self._aConnection, formats=self._formats,
-                send_to_device=self._send_to_device, create_xray=self._create_xray, proxy=self._proxy,
-                https_address=self._https_address, https_port=self._https_port)
-            self._books.append(book)
-            self._gConnection = book._gConnection
-            self._aConnection = book._aConnection
+            self._books.append(Book(self._db, book_id, self._gConnection, self._aConnection, formats=self._formats,
+                send_to_device=self._send_to_device, create_xray=self._create_xray))
         
         self._total_not_failing = 0
         book_lookup = {}
```

Comment:
The cleanup you added looks good, but you didn't do it all (you got rid of the unnecessary return values, but eg. we are still passing through `gConnection` when calling `book.create_xray_event` - when it could be cached at init time.

You might have had a reason for not removing that though, so just calling it out.